libraries/AC_PrecLand/AC_PrecLand.cpp

```diff
@@ -114,6 +114,40 @@ const AP_Param::GroupInfo AC_PrecLand::var_info[] = {
     // @RebootRequired: True
     AP_GROUPINFO("LAG", 9, AC_PrecLand, _lag, 0.02f), // 20ms is the old default buffer size (8 frames @ 400hz/2.5ms)
 
+    // @Param: STRICT
+    // @DisplayName: PrecLand strictness
+    // @Description: How strictly should the vehicle land on the target if target is lost
+    // @Values: 0: Land Vertically, 1: Retry Landing, 2: Do not land(Hover)
+    AP_GROUPINFO("STRICT", 10, AC_PrecLand, _strict, 1),
+
+    // @Param: RT_MAX
+    // @DisplayName: PrecLand Maximum number of retires to a failed landing
+    // @Description: PrecLand Maximum number of retires to a failed landing. Set zero for no retry.
+    // @Range: 0 10
+    // @Increment: 1
+    AP_GROUPINFO("RT_MAX", 11, AC_PrecLand, _max_retry_num, 4),
+
+    // @Param: RT_TIMEOUT
+    // @DisplayName: PrecLand retry timeout
+    // @Description: Time for which vehicle continues descenind even if target is lost.
+    // @Range: 0 20
+    // @Units: s
+    AP_GROUPINFO("RT_TIMEOUT", 12, AC_PrecLand, _retry_time_start_sec, 4),
```

Comment:
Hum, I got the same issue in my implementation ... This can be very long depending the descend speed. we may need some sort of checker to unsure that we won't crash before timeout